lib/Logger.js

```diff
@@ -55,7 +77,9 @@ module.exports = class Logger {
 
   _output(level, record) {
     this._transports.forEach((transport) => {
-      transport(level, record);
+      const p = transport(level, record) || Promise.resolve();
+      this._promises.add(p);
```

Comment:
This Set could get infinitely large if the transport does not yield fast enough. An option would be to set a max threshold and to ignore promises after this had been breached, but I don't think it's a good solution since it would hide the slow transport. I think it's better to fail fast by running out of memory if the transport is too slow

Reply:
Yes, fix this issue if/when the problem surfaces.